azurelinuxagent/common/protocol/extensions_goal_state_from_vm_settings.py

```diff
@@ -100,15 +100,21 @@ def _parse_vm_settings(self, json_text):
         # TODO: Parse all atttributes
 
     def _parse_simple_attributes(self, vm_settings):
+        self._activity_id = self._string_to_id(vm_settings.get("activityId"))
+        self._correlation_id = self._string_to_id(vm_settings.get("correlationId"))
+        self._created_on_timestamp = self._ticks_to_utc_timestamp(vm_settings.get("extensionsLastModifiedTickCount"))
+
         host_ga_plugin_version = vm_settings.get("hostGAPluginVersion")
         if host_ga_plugin_version is not None:
             self._host_ga_plugin_version = FlexibleVersion(host_ga_plugin_version)
+
         schema_version = vm_settings.get("vmSettingsSchemaVersion")
         if schema_version is not None:
             self._schema_version = FlexibleVersion(schema_version)
-        self._activity_id = self._string_to_id(vm_settings.get("activityId"))
-        self._correlation_id = self._string_to_id(vm_settings.get("correlationId"))
-        self._created_on_timestamp = self._ticks_to_utc_timestamp(vm_settings.get("extensionsLastModifiedTickCount"))
+
```

Comment:
Will it be a good idea to add an comment about what "onHold" is as you have explained in the chat ? - "if overprovisioning is enabled, when creating a scale set, a few extra machines are created and they are put "on hold" in case some of the other machines fail"

I understand we can always refer to other documentation

Reply:
Thanks for the suggestion, but explaining that in the code at this point is just too much. Do we need to explain what "onHold" is? what the WireServer is? What Fabric is?, etc. I'd suggest asking in the internal chat or checking the internal documentation for that level of detail. Do feel free to ask in PRs (or the internal chat) when something is not clear to you.

Reply:
Yes agreed